lark/grammars/python.lark

```diff
@@ -225,17 +225,18 @@ _NEWLINE: ( /\r?\n[\t ]*/ | COMMENT )+
 NAME: /[a-zA-Z_]\w*/
 COMMENT: /#[^\n]*/
 
-STRING : /[ubf]?r?("(?!"").*?(?<!\\)(\\\\)*?"|'(?!'').*?(?<!\\)(\\\\)*?')/i
+STRING: /[ubf]?r?("(?!"").*?(?<!\\)(\\\\)*?"|'(?!'').*?(?<!\\)(\\\\)*?')/i
 LONG_STRING: /[ubf]?r?(""".*?(?<!\\)(\\\\)*?"""|'''.*?(?<!\\)(\\\\)*?''')/is
 
-DEC_NUMBER: /0|[1-9][\d_]*/i
-HEX_NUMBER.2: /0x[\da-f]*/i
-OCT_NUMBER.2: /0o[0-7]*/i
-BIN_NUMBER.2 : /0b[0-1]*/i
-FLOAT_NUMBER.2: /((\d+\.[\d_]*|\.[\d_]+)([Ee][-+]?\d+)?|\d+([Ee][-+]?\d+))/
-IMAG_NUMBER.2: /\d+[Jj]/ | FLOAT_NUMBER /[Jj]/
+DEC_NUMBER: "0"+ | "1".."9" ("_"?  "0".."9"                       )*
+HEX_NUMBER.2: "0x"          ("_"? ("0".."9" | "a".."f" | "A".."F"))+
+OCT_NUMBER.2: "0o"          ("_"?  "0".."7"                       )+
+BIN_NUMBER.2: "0b"          ("_"?  "0".."1"                       )+
+FLOAT_NUMBER.2: (DEC_NUMBER "." DEC_NUMBER? | "." DEC_NUMBER) (("E" | "e") ["+" | "-"] DEC_NUMBER)?
```

Comment:
there's no lower/upper case for numbers(?), so the only thing we need to take from `/.../i` is in the hex, where we have `a-f`